spirv_glsl.cpp

```diff
@@ -3542,6 +3542,7 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 			require_extension_internal("GL_ARB_shader_texture_lod");
 	}
 
+	// texture
 	if (op == "texture")
```

Comment:
Isn't it easier to move the check for "texture" vs "shadow" here? You know the imgtype here, so you should be able to isolate the check to think function.